manim/utils/simple_functions.py

```diff
@@ -88,10 +88,19 @@ def choose(n: int, k: int) -> int:
     - https://en.wikipedia.org/wiki/Combination
     - https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.comb.html
     """
-    return special.comb(n, k, exact=True)
+    value: int = special.comb(n, k, exact=True)
+    return value
 
 
-def clip(a, min_a, max_a):
+@overload
+def clip(a: float, min_a: float, max_a: float) -> float: ...
+
+
+@overload
+def clip(a: str, min_a: str, max_a: str) -> str: ...
+
+
+def clip(a, min_a, max_a):  # type: ignore[no-untyped-def]
```

Comment:
Instead of doing the overloads, my suggestion is to create a `Comparable` protocol which supports the `__lt__()` (less than) and `__gt__()` (greater than) dunder methods and then define a `ComparableT` TypeVar:
```suggestion
class Comparable(Protocol):
    def __lt__(self, other: Any) -> bool: ...
    
    def __gt__(self, other: Any) -> bool: ...

ComparableT = TypeVar("ComparableT", bound=Comparable)  # noqa: Y001
    
def clip(a: ComparableT, min_a: ComparableT, max_a: ComparableT) -> ComparableT:
```